### PR TITLE
APP-611 origin check multiple ips

### PR DESCRIPTION
**Pull request by ecarrenho-daitan**
* When the x-forwarded-for header contains multiple IP's separated by comma, the request will be accepted if any of the IP's is listed the the allowed origins list.
* Log message text added to resource bundle.
* Exceptions using varargs list of strings.

### CODE REVIEW

File: integration-web/src/main/java/org/symphonyoss/integration/web/filter/WebHookOriginCheckFilter.java [hunk 1]
```diff
@@ -115,20 +151,23 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
...
-      String remoteAddress = getRemoteAddress(request);
-      boolean allowedOrigin = verifyOrigin(remoteAddress, whiteList);
+      String remoteAddressInfo = getOriginatingAddressInfo(request);
+      boolean allowedOrigin = verifyOrigin(remoteAddressInfo, whiteList, integrationType);
 
       if (allowedOrigin) {
         filterChain.doFilter(servletRequest, servletResponse);
       } else {
-        writeResponse(response, whiteList, remoteAddress);
+        LOGGER.warn(ExceptionMessageFormatter.format("Webhook Filter",
```
Comment (robson-sanchez) [inline integration-web/src/main/java/org/symphonyoss/integration/web/filter/WebHookOriginCheckFilter.java]: "Webhook Filter" should be a constant field to reuse in the class.

**Review by robson-sanchez — APPROVED** [review]
Minor changes.

In general, I think we should put the error and warning messages in a separate properties file like a message source. I believe it should be better than define big constants.

Example: https://www.mkyong.com/spring/spring-resource-bundle-with-resourcebundlemessagesource-example/

It doesn't need to be implemented now, but we should consider in the next sprints.

File: integration-web/src/main/java/org/symphonyoss/integration/web/filter/WebHookOriginCheckFilter.java (hunk 1 above)
Comment (ecarrenho-daitan) [reply integration-web/src/main/java/org/symphonyoss/integration/web/filter/WebHookOriginCheckFilter.java]: Done.

File: integration-web/src/test/java/org/symphonyoss/integration/web/filter/WebHookOriginCheckFilterTest.java [hunk 2]
```diff
@@ -60,6 +61,12 @@
 
   private static final String REMOTE_ADDRESS = "192.30.252.40";
 
+  private static final String REMOTE_ADDRESS_LIST_FIRST_IP_ALLOWED = "192.30.252.40, 168.140.252.55";
+
+  private static final String REMOTE_ADDRESS_LIST_SECOND_IP_ALLOWED = "10.30.224.40, 192.30.252.40";
```
Comment (dnsymphony) [inline integration-web/src/test/java/org/symphonyoss/integration/web/filter/WebHookOriginCheckFilterTest.java:66]: One of these should verify that the regex will work for IPs that are separated by comma but no space.

**Review by dnsymphony — CHANGES_REQUESTED** [review]

File: integration-web/src/test/java/org/symphonyoss/integration/web/filter/WebHookOriginCheckFilterTest.java (hunk 2 above)
Comment (ecarrenho-daitan) [reply integration-web/src/test/java/org/symphonyoss/integration/web/filter/WebHookOriginCheckFilterTest.java:66]: Done.

**Pull request merged**